src/S3/PostObjectV4.php

```diff
@@ -56,7 +56,7 @@ public function __construct(
         $credentials   = $this->client->getCredentials()->wait();
         $securityToken = $credentials->getSecurityToken();
 
-        if (null !== $securityToken) {
+        if (!empty($securityToken)) {
```

Comment:
`if (!empty($variable))` is shorthand for `if (isset($variable) && $variable)`. Both consider the condition met if `$variable` contains a truthy value and fail if it contains falsy one (including an empty string).